server/modules/amazon-client/s3Url.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @typedef {Object} S3Location
+ * @property {string} endpoint
+ * @property {string} Bucket
+ * @property {string} Key
+ * @property {string|undefined} VersionId
+ */
+
+'use strict';
+
+const AWS = require('aws-sdk');
+const _ = require('lodash/fp');
+
+/**
+ * Parse a string as an S3 object URL.
+ * @param {string} url - the URL that refers to the S3 object.
+ * @returns {S3Location|undefined} The parsed S3 location.
+ */
+function parse(url) {
+  let regex = /^(https:\/\/s3[^\/\.]+\.amazonaws\.com)\/([^\/]+)\/([^\?]+)(?:\?versionId=([^&]+))?$/;
+  let t = regex.exec(url);
```

Comment:
Do we have examples of what we expect to match (or exclude) with this? Do we always require a version Id?

Do we support [Virtual Host S3 Urls](http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html)? :
```
https://s3.amazonaws.com/johnsmith.net/homepage.html?versionId=234234343
```